CMakeLists.txt

```diff
@@ -12200,44 +12199,6 @@ target_link_libraries(interop_server
 )
 
 
-endif()
-if(gRPC_BUILD_TESTS)
-if(_gRPC_PLATFORM_LINUX OR _gRPC_PLATFORM_MAC OR _gRPC_PLATFORM_POSIX)
-
-  add_executable(interop_test
-    test/cpp/interop/interop_test.cc
-    third_party/googletest/googletest/src/gtest-all.cc
-    third_party/googletest/googlemock/src/gmock-all.cc
-  )
-
-  target_include_directories(interop_test
```

Comment:
any reason why interop_test be no longer buildable with cmake? Or is that an oversight?

Reply:
I'll add that I think this would be a special case, the only binary in a test folder that would need special handling because no exported tests depend on it. I didn't do anything to explicitly exclude it, per se.

Reply:
well something in this PR clearly excludes this target from the cmake build and I think it's important to understand what caused it. We can't just make changes that have side effects without understanding what is causing those side effects and having a discussion whether these side effects make sense or not.
I pointed out this one since it seemed that removing interop_test target from cmake has nothing to do with the purpose of this change, so to me it's a red flag that the PR does something we don't quite understand.

So I don't feel strongly about keeping interop_test as part of cmake build (probably not a disaster if it is removed), but I'd like to see that effort has been put in keeping it there and that we have a good understanding why it disappeared.

Reply:
> We can't just make changes that have side effects without understanding what is causing those

You're right to point it out, absolutely, and having a fruitful discussion is important. 

> to me it's a red flag that the PR does something we don't quite understand.

Not the case, the mechanics are clear. This target was a unique case beforehand, that's not new. It's plausible that this target was added into cmake accidentally in the first place, though I'm not sure our CI would be sensitive to that (our master builds would, at least). `interop_client` and `interop_server` still exist in the cmake build, which is important because I _did_ find uses of them via cmake in our master builds. If this target is relied upon via cmake, it's not obvious where or how. Putting in extra effort to keep what _may be_ a mistake is not the right move in my opinion. If some tests break due to non-obvious usages, that's a problem that this PR would help identify. 

All in all, a clear improvement is presented here, a best effort was made, our builds and various test environments are complex, and in any case rollbacks happen frequently on this repo in the normal course of operation due to issues that can be difficult to predict. I'd be happy to be shown it's needed, and put in the work required to document that somewhere obvious, and handle this edge case in the cmake export.